chapter2/exercise4.js

```diff
@@ -0,0 +1,37 @@
+(function() {
+
+    displayFlexChessBoard = function(widthChessBoard, heightChessBoard) {
+        var lineOfChessBoard = "", subStringLineOfChessBoard = "";
+
+        if((typeof widthChessBoard == "undefined") || (typeof heightChessBoard == "undefined")){
+            return console.log("You don't pass needed arguments to fucntion.");
+        } else {
+            for(var positionInString = 0; positionInString < widthChessBoard; positionInString++) {
+                if((positionInString % 2) == 0) {
+                    lineOfChessBoard += " ";
+                } else {
+                    lineOfChessBoard += "#";
+                }
+            }
+            
+            for(var rowOfChessBoard = 0; rowOfChessBoard < heightChessBoard; rowOfChessBoard++){           
+                if(rowOfChessBoard == 0){
+                    console.log(lineOfChessBoard);
+                } else if((rowOfChessBoard != 0) && ((rowOfChessBoard % 2) == 0)) {
+                    subStringLineOfChessBoard = lineOfChessBoard.substring(0, lineOfChessBoard.length - 1);
+                    lineOfChessBoard = " ";
+                    lineOfChessBoard += subStringLineOfChessBoard;
+                    console.log(lineOfChessBoard);
+                } else if ((rowOfChessBoard != 0) && ((rowOfChessBoard % 2) != 0)) {
+                    subStringLineOfChessBoard = lineOfChessBoard.substring(0, lineOfChessBoard.length - 1);
```

Comment:
linie 26-29 sa wlasciwie takie same jak 21-24. sadze, ze nie ma wiec sensu, zeby sie powtarzaly. przepisz to prosze tak, zebys nie mial dwa razy tego samego kodu.

Reply:
Poprawiłem całkowicie zadanie. Poprzednie, w zły sposób wyświetlało szachownice oraz w zły sposób zaczynało ją wyświetlać.

Reply:
I kod strasznie nie podobał mi się. Był bezsensu. Można to było prościej zrobić.